### PR TITLE
Fix Union ctor to make compare function usable

### LINKED ISSUE #427: Union ctor with function is corrupt
### Expected Behavior
We have a ctor with a function which accepts two input arguments like this:
`Union(IEnumerable<T> a, IEnumerable<T> b, Func<T, T, bool> compare)`

### Actual Behavior
The only ctor which accepts a function is this:
`Union(IEnumerable<T> a, IEnumerable<T> b, Func<T, bool> compare)`.
The compare function has no access to both items to compare. Thus it is not usable and has to be replaced.

### The log given by the failure.

### PR DESCRIPTION
**Pull request by aveXcaesar**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)

closes #427 

### What is the new behavior?

Union with compare function: `Union(IEnumerable<T> a, IEnumerable<T> b, Func<T, T, bool> compare)`

### Does this PR introduce a breaking change? (check one with "x")
- [x] Yes
- [ ] No

### If this PR contains a breaking change, please describe the impact and migration path for existing applications
Change ctor from
`Union(IEnumerable<T> a, IEnumerable<T> b, Func<T, bool> compare)`
to
`Union(IEnumerable<T> a, IEnumerable<T> b, Func<T, T, bool> compare)`
###  Other information
I can't imagine any use case for the refactored ctor `Union(IEnumerable<T> a, IEnumerable<T> b, Func<T, bool> compare)`.
If there is any I can undo deleting this and thus remove the breaking change.
What do you think?

### CODE REVIEW

File: src/Yaapii.Atoms/Enumerable/Union.cs [hunk 1]
```diff
@@ -30,20 +30,22 @@ namespace Yaapii.Atoms.Enumerable
...
-                    if (new Contains<T>(a, compare).Value())
-                    {
-                        result.Add(item);
-                    }
-                }
-                return result;
+        /// <summary>
+        /// Union objects in two enumerables.
+        /// </summary>
+        /// <param name="compare">Decission to put items from a in the resulting union</param>
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/Enumerable/Union.cs:36]:
```suggestion
        /// <param name="compare">Condition for to match</param>
```

File: tests/Yaapii.Atoms.Tests/Enumerable/UnionTests.cs [hunk 2]
```diff
@@ -63,5 +66,26 @@ public void MatchesInt(IEnumerable<int> a, IEnumerable<int> b, IEnumerable<int>
                 )
             );
         }
+
+        [Fact]
+        public void UsesCompareFunction()
+        {
+            var a = new ListOf<string>("c:/abraham/a.jpg", "c:/bertram/b.jpg", "c:/caesar/c.jpg");
+            var b = new ListOf<string>("a", "c");
+            var expected = new ListOf<string>("c:/abraham/a.jpg", "c:/caesar/c.jpg");
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/Enumerable/UnionTests.cs:75]: plz put this into line 78 as this is unnecessary temporal coupling.

File: tests/Yaapii.Atoms.Tests/Enumerable/UnionTests.cs [hunk 3]
```diff
@@ -63,5 +66,26 @@ public void MatchesInt(IEnumerable<int> a, IEnumerable<int> b, IEnumerable<int>
                 )
             );
         }
+
+        [Fact]
+        public void UsesCompareFunction()
+        {
+            var a = new ListOf<string>("c:/abraham/a.jpg", "c:/bertram/b.jpg", "c:/caesar/c.jpg");
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/Enumerable/UnionTests.cs:73]: plz put this into line 80 as this is unnecessary temporal coupling.

File: tests/Yaapii.Atoms.Tests/Enumerable/UnionTests.cs [hunk 4]
```diff
@@ -63,5 +66,26 @@ public void MatchesInt(IEnumerable<int> a, IEnumerable<int> b, IEnumerable<int>
...
+            var b = new ListOf<string>("a", "c");
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/Enumerable/UnionTests.cs:74]: plz put this into line 81 as this is unnecessary temporal coupling.

File: src/Yaapii.Atoms/Enumerable/Union.cs [hunk 5]
```diff
@@ -30,20 +30,22 @@ namespace Yaapii.Atoms.Enumerable
...
+        public Union(IEnumerable<T> a, IEnumerable<T> b, Func<T, T, bool> compare) : base(() =>
+            {
+                var result = new List<T>();
+                foreach (var aItem in a)
+                {
+
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/Enumerable/Union.cs:42]: plz remove

**Review by Meerownymous — CHANGES_REQUESTED** [review]

**Comment by aveXcaesar** [comment]
@Meerownymous 
It's done

**Comment by aveXcaesar** [comment]
@koeeenig 
please have a look and merge the pr if possible.

**Review by koeeenig — APPROVED** [review]

**Comment by aveXcaesar** [comment]
@koeeenig @Meerownymous 
please merge...

**Pull request merged**